src/orange/surf/Involute.hh

```diff
@@ -0,0 +1,305 @@
+ * the origin.
+ *
+ * Next the point is determined to be inside if an involute that passes it,
+ * has an \em t that is within the bounds and an \em a greater than the
+ * involute being tested for. To do this, the tangent to the involute must be
+ * determined by calculating the intercepts of the circle of involute and
+ * circle centered at the midpoint of the origin and the point being tested for
+ * that extends to the origin. After which the angle of tangent point can be
+ * determined and used to calculate \em a via: \f[
+ * a = angle - t_point
+ * \f]
+ *
+ * To calculate the intercept of two circles it is assumed that the two circles
+ * are \em x_prime axis. The coordinate along \em x_prime originating from the
+ * center of one of the circles is given by: \f[ x_prime = (d^2 - r^2 +
+ * R^2)/(2d) \f] Where \em d is the distance between the center of the two
+ * circles, \em r is the radius of the displaced circle, and \em R is the
+ * radius of the circle at the orgin. Then the point \em y_prime can be
+ * obatined from: \f[ y_prime = \pm sqrt(R^2 - x_prime^2) \f] Then to convert (
+ * \em x_prime , \em y_prime ) to ( \em x, \em y ) the following rotation is
+ * applied:\f[ x = x_prime*cos(theta) - y_prime*sin(theta) y =
+ * y_prime*cos(theta) + x_prime*sin(theta) \f] Where \em theta is the angle
+ * between the \em x_prime axis and the \em x axis.
+ */
+CELER_FUNCTION SignedSense Involute::calc_sense(Real3 const& pos) const
+{
+    using constants::pi;
+
+    real_type x = pos[0] - origin_[0];
+    real_type const y = pos[1] - origin_[1];
+
+    if (this->sign() == Sign::clockwise)
+    {
+        x = negate(x);
+    }
+
+    // Calculate distance to origin and obtain t value for distance.
+    real_type const t_point_sq = (ipow<2>(x) + ipow<2>(y) / ipow<2>(r_b_)) - 1;
+
+    // Check if point is in defined bounds.
+    if (t_point_sq < ipow<2>(tmin_))
+    {
+        return SignedSense::outside;
+    }
+    if (t_point_sq > ipow<2>(tmax_))
+    {
+        return SignedSense::outside;
+    }
+
+    // Check if Point is on involute.
+    detail::InvolutePoint calc_point{this->r_b(), a_};
+    Real2 point = calc_point(clamp_to_nonneg(t_point_sq));
+
+    if (x == point[0] && y == point[1])
+    {
+        return SignedSense::on;
+    }
+
+    // Check if point is in interval
+
+    // Calculate tangent point
+    real_type x_prime = ipow<2>(r_b_) / std::sqrt(ipow<2>(x) + ipow<2>(y));
+    real_type y_prime = std::sqrt(ipow<2>(r_b_) - ipow<2>(x_prime));
+
+    point[0] = (x_prime * x - y_prime * y) / std::sqrt(ipow<2>(x) + ipow<2>(y));
+    point[1] = (y_prime * x + x_prime * y) / std::sqrt(ipow<2>(x) + ipow<2>(y));
+
+    // Calculate angle of tangent
+    real_type theta = std::acos(point[0] / norm(point));
+    if (point[1] < 0)
+    {
+        theta = 2 * pi - theta;
+    }
+    theta += max<real_type>(0, std::floor((tmax_ + a_ - theta) / (2 * pi))) * 2
+             * pi;
```

Comment:
So here, the range of theta starts out in `[0, π]`, then is `[0, 2π]`. tmax is in `[0, 4π)`, a is in `[0, 2π)` maybe, so
`tmax_ + a_ - theta` is in `[-2π, 6π)`.  If it's negative, `std::floor` returns the more negative multiple (so -π gets rounded to -2π and then `max`ed to zero).  So I think this is really just
```c+++
theta += max(real_type{0}, std::fmod(tmax_ + a_ - theta, 2 * pi));
```

I think a little extension of the comment above it could help readability too.